properties/base.py

```diff
@@ -135,51 +169,42 @@ def __new__(mcs, name, bases, classdict):                                  #pyli
 class HasProperties(with_metaclass(PropertyMetaclass, object)):
     """HasProperties class with properties"""
 
-    _backend_name = "dict"
-    _backend_class = dict
-    _defaults = None
+    _defaults = dict()
     _REGISTRY = dict()
 
     def __init__(self, **kwargs):
-        self._backend = self._backend_class()
-
-        # add the default listeners
+        self._backend = dict()
         self._listeners = dict()
+
+        # Register the listeners
         for _, val in iteritems(self._prop_observers):
             handlers._set_listener(self, val)
 
-        # set the defaults
-        defaults = self._defaults or dict()
-        for key, value in iteritems(defaults):
-            if key not in self._props.keys():
-                raise AttributeError(
-                    "Default input '{:s}' is not a known property".format(key)
-                )
-            if callable(value):
-                setattr(self, key, value())
-            else:
-                setattr(self, key, value)
-
-        # set the keywords
-        for key in kwargs:
+        # Set the GettableProperties from defaults - these are only set here
+        for key in self._props:
+            if not isinstance(self._props[key], basic.Property):
+                if key in self._defaults:
+                    val = self._defaults[key]
+                else:
+                    val = self._props[key].default
+                if val is utils.undefined:
+                    continue
+                if callable(val):
+                    val = val()
+                self._props[key].validate(self, val)
+                self._backend[key] = val
+
+        # Set the other defaults without triggering change notifications
+        self.reset(silent=True)
+
+        # Set the keyword arguments with change notifications
+        for key, val in iteritems(kwargs):
             if not hasattr(self, key) and key not in self._props.keys():
-                raise AttributeError("Keyword input '{:s}' is not a known "
+                raise AttributeError("Keyword input '{}' is not a known "
                                      "property or attribute".format(key))
-            setattr(self, key, kwargs[key])
+            setattr(self, key, val)
```

Comment:
@fwkoch: You were talking about maybe moving this stuff to `__new__`. Another option would be to move it to the metaclass `__call__`, as in [galoshes](https://github.com/bsmithyman/galoshes/blob/master/galoshes/meta.py#L60). That leaves the `__init__.py` undefined, and free to do whatever it wants, within a fully-initialized class. Thoughts?